### PR TITLE
refactor(core): Move `typeorm` operators from `WaitTracker` to `ExecutionRepository` (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Follow-up to: #8145

### CODE REVIEW

File: packages/cli/src/databases/repositories/execution.repository.ts [hunk 1]
```diff
@@ -516,4 +517,29 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 	async deleteByIds(executionIds: string[]) {
 		return this.delete({ id: In(executionIds) });
 	}
+
+	async getWaitingExecutions() {
```
Comment (netroy) [inline packages/cli/src/databases/repositories/execution.repository.ts:521]: should we add unit tests for this to cover for the `sqlite` check ?

File: packages/cli/test/unit/repositories/execution.repository.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+describe('ExecutionRepository', () => {
+	const entityManager = mockInstance(EntityManager);
+	const dataSource = mockInstance(DataSource, { manager: entityManager });
+	dataSource.getMetadata.mockReturnValue(mock());
+	Object.assign(entityManager, { connection: dataSource });
+
+	const executionRepository = Container.get(ExecutionRepository);
+
+	describe('getWaitingExecutions()', () => {
+		test('should', async () => {
```
Comment (netroy) [inline packages/cli/test/unit/repositories/execution.repository.test.ts:16]: can we also add tests to cover `dbType === 'sqlite'` and `dbType !== 'sqlite'` cases?
You could mock config using `jest.spyOn(config, 'getEnv')`

File: packages/cli/test/unit/repositories/execution.repository.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+				await executionRepository.getWaitingExecutions();
+
+				const expectedQuery = expect.objectContaining({
+					order: { waitTill: 'ASC' },
+					select: ['id', 'waitTill'],
+					where: expect.objectContaining({
+						waitTill: expect.objectContaining({
+							_type: 'lessThanOrEqual',
+							_value: expect.any(String),
```
Comment (netroy) [inline packages/cli/test/unit/repositories/execution.repository.test.ts:31]: This unfortunately does not cover that for `sqlite` we call `DateUtils.mixedDateToUtcDatetimeString`.

**Review by netroy — APPROVED** [review]
👍🏽

**Comment by Joffcom** [comment]
Got released with [`n8n@1.23.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.23.0)

**Pull request merged**